tests/extensions/shimmy/shimmy.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+require_once 'shimmy.civix.php';
+// phpcs:disable
+use CRM_Shimmy_ExtensionUtil as E;
+// phpcs:enable
+
+/**
+ * Implements hook_civicrm_config().
+ *
+ * @link https://docs.civicrm.org/dev/en/latest/hooks/hook_civicrm_config/
+ */
+function shimmy_civicrm_config(&$config) {
+  _shimmy_civix_civicrm_config($config);
+}
+
+/**
+ * Implements hook_civicrm_install().
+ *
+ * @link https://docs.civicrm.org/dev/en/latest/hooks/hook_civicrm_install
+ */
+function shimmy_civicrm_install() {
+  _shimmy_civix_civicrm_install();
+}
+
+/**
+ * Implements hook_civicrm_postInstall().
+ *
+ * @link https://docs.civicrm.org/dev/en/latest/hooks/hook_civicrm_postInstall
+ */
+function shimmy_civicrm_postInstall() {
+  _shimmy_civix_civicrm_postInstall();
+}
+
+/**
+ * Implements hook_civicrm_uninstall().
+ *
+ * @link https://docs.civicrm.org/dev/en/latest/hooks/hook_civicrm_uninstall
+ */
+function shimmy_civicrm_uninstall() {
+  _shimmy_civix_civicrm_uninstall();
+}
+
+/**
+ * Implements hook_civicrm_enable().
+ *
+ * @link https://docs.civicrm.org/dev/en/latest/hooks/hook_civicrm_enable
+ */
+function shimmy_civicrm_enable() {
+  _shimmy_civix_civicrm_enable();
+}
+
+/**
+ * Implements hook_civicrm_disable().
+ *
+ * @link https://docs.civicrm.org/dev/en/latest/hooks/hook_civicrm_disable
+ */
+function shimmy_civicrm_disable() {
+  _shimmy_civix_civicrm_disable();
+}
+
+/**
+ * Implements hook_civicrm_upgrade().
+ *
+ * @link https://docs.civicrm.org/dev/en/latest/hooks/hook_civicrm_upgrade
+ */
+function shimmy_civicrm_upgrade($op, CRM_Queue_Queue $queue = NULL) {
+  return _shimmy_civix_civicrm_upgrade($op, $queue);
+}
+
+/**
+ * Implements hook_civicrm_entityTypes().
+ *
+ * Declare entity types provided by this module.
+ *
+ * @link https://docs.civicrm.org/dev/en/latest/hooks/hook_civicrm_entityTypes
+ */
+function shimmy_civicrm_entityTypes(&$entityTypes) {
+  _shimmy_civix_civicrm_entityTypes($entityTypes);
+}
```

Comment:
Seems a bit ironic that the shimmy extension has a civix file implementing the boilerplate that the extension is designed to get rid of :)

Reply:
Heh :) Yeah, the process I followed was basically:

1. Make an extension (`civix generate:module shimmy`) with lots of boilerplate functions.
2. Read through the boilerplate functions.
3. One-by-one, delete each function and write a mixin. (Most of these were cookie-cutter changes.)

The leftover file is like a TODO list ("Things to reorganize!"). A lot of the remaining boilerplate can be nixed, but they're not quite so cookie-cutter.

* Install/enable/upgrade stuff should use `<upgrader>`+`<classloader>` instead of `<mixin>`.
* The `include_path` and `entityTypes` stuff should be amenable to moving to a mixin, assuming we want to keep those code-patterns.
* The `insert_navigation_menu()` stuff is wonky. We can probably delete the `fixNavigationMenu` straight-up.
* That leaves `CRM_*_ExtensionUtil` and `_*_civix_mixin_polyfill` as things to preserve for the long term.